src/dashboard/src/components/helpers.py

```diff
@@ -300,32 +303,81 @@ def processing_config_path():
+    mets_path = "{}/data/{}".format(absolute_transfer_name, mets_name)
+    mets_tmp_dir = tempfile.mkdtemp()
+    mets_tmp_file = os.path.join(mets_tmp_dir, mets_name)
+    # We can't get a lot of debug information from AMClient yet, so we try to
+    # download and then open, returning an error if the file can't be accessed.
+    try:
+        AMClient(
+            ss_api_key=get_setting("storage_service_apikey", None),
+            ss_user_name=get_setting("storage_service_user", "test"),
+            ss_url=get_setting("storage_service_url", None).rstrip("/"),
+            package_uuid=sip_uuid,
+            relative_path=mets_path,
+            saveas_filename=mets_tmp_file,
+        ).extract_file()
+    except requests.exceptions.ConnectionError as err:
+        response = {"success": False, "message": error_message.format(err)}
+        return json_response(response, status_code=503)
+    mets = ""
+    err_msg = None
+    try:
+        with open(mets_tmp_file, "r") as mets_file:
+            mets = mets_file.read()
+        os.unlink(mets_tmp_file)
+    except IOError as err:
+        err_msg = err
+    if not mets:
+        response = {"success": False, "message": error_message.format(err_msg)}
+        return json_response(response, status_code=404)
+    content_type = "application/xml"
+    content_disposition = "attachment; filename={};".format(mets_name)
+    return _prepare_stream_response(
+        payload=mets, content_type=content_type, content_disposition=content_disposition
+    )
+
+
 def stream_file_from_storage_service(
     url, error_message="Remote URL returned {}", preview_file=False
 ):
     # Repetetive or constant values to use below when seeting the headers.
-    header_content_disposition = "Content-Disposition"
-    content_disposition_inline = "inline"
     storage_timeout = django_settings.STORAGE_SERVICE_CLIENT_TIMEOUT
     stream = requests.get(url, stream=True, timeout=storage_timeout)
-    if stream.status_code == 200:
-        content_type = stream.headers.get("content-type", "text/plain")
-        response = StreamingHttpResponse(stream, content_type=content_type)
-
-        # Provide a content-disposition so the browser knows how to handles
-        # the response.
-        content_disposition = stream.headers.get("content-disposition")
-        if preview_file:
-            response[header_content_disposition] = content_disposition_inline
-        elif content_disposition:
-            response[header_content_disposition] = content_disposition
-        return response
-    else:
+    if stream.status_code != 200:
         response = {
             "success": False,
             "message": error_message.format(stream.status_code),
         }
         return json_response(response, status_code=400)
+    content_type = stream.headers.get("content-type", "text/plain")
+    content_disposition = stream.headers.get("content-disposition")
+    return _prepare_stream_response(
+        payload=stream.content,
```

Comment:
Hi @ross-spencer, I wonder if this is breaking the stream flow and loading the entire content on memory to re-stream it later.

Reply:
I think I see what you mean @jraddaoui with the initial get requesting a stream any suggestions how to resolve? Or test? I can revert the behavior as well, but what do you think?

Reply:
For reference, the original function: is [here](https://github.com/artefactual/archivematica/blob/1a47b3f9227eef8e8d7a9901215fdeab52423977/src/dashboard/src/components/helpers.py#L311-L328)

Reply:
I'd try to implement `_prepare_stream_response` using a stream instead of the full content. And I'd add a `stream_file` function to AMClient (as noted below) from where you can get a stream and pass it to `_prepare_stream_response` in a similar manner.

Reply:
Hi @jraddaoui as discussed in Slack how is this for an initial compromise? The flag allows us to not stream for the METS while we develop out AMClient, and once that functionality is available we'll be able to switch that flag and remove the code that relies on it. Tests now reflect the data we're anticipating better.

The commit is: 7ac55a035b771c059e365a462aa5954e4cfb8b3d

Reply:
Very glad that @jraddaoui spotted this :bowing_man: I've been thinking this was code just being extracted into a new method, and didn't notice the change in the behaviour.